### PR TITLE
Add default CA rule enablement info

### LINKED ISSUE #32954: CAxxxx pages should mention whether a rule is disabled by default.
`dotnet_analyzer_diagnostic.category-<category>.severity` syntax doesn't apply to rules that are disabled by default. So it is beneficial to know in each page whether a rule is disabled by default or not.

Those pages should probably not mention `dotnet_analyzer_diagnostic.category-<category>.severity` at all to avoid confusion?

See https://github.com/dotnet/roslyn-analyzers/issues/6188

---
[Associated WorkItem - 57954](https://dev.azure.com/msft-skilling/Content/_workitems/edit/57954)

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #32954

- Adds default enablement info to overview table of each rule
- Removes configuration-by-category info if rule is disabled by default

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/code-analysis/quality-rules/ca1002.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1002.md) | [CA1002: Do not expose generic lists](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1005.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1005.md) | [CA1005: Avoid excessive parameters on generic types](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1005?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1010.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1010.md) | [CA1010: Collections should implement generic interface](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1010?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1012.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1012.md) | [CA1012: Abstract types should not have public constructors](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1012?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1014.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1014.md) | [CA1014: Mark assemblies with CLSCompliantAttribute](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1014?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1016.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1016.md) | [CA1016: Mark assemblies with AssemblyVersionAttribute](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1016?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1019.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1019.md) | [CA1019: Define accessors for attribute arguments](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1019?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1021.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1021.md) | [CA1021: Avoid out parameters](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1021?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1024.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1024.md) | [CA1024: Use properties where appropriate](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1024?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1027.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1027.md) | [CA1027: Mark enums with FlagsAttribute](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1027?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1028.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1028.md) | [CA1028: Enum storage should be Int32](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1028?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1030.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1030.md) | [CA1030: Use events where appropriate](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1030?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1031.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1031.md) | [CA1031: Do not catch general exception types](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1031?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1032.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1032.md) | [CA1032: Implement standard exception constructors](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1032?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1033.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1033.md) | [CA1033: Interface methods should be callable by child types](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1033?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1036.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1036.md) | [CA1036: Override methods on comparable types](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1036?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1040.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1040.md) | [CA1040: Avoid empty interfaces](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1040?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1043.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1043.md) | [CA1043: Use integral or string argument for indexers](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1043?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1045.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1045.md) | [CA1045: Do not pass types by reference](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1045?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1046.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1046.md) | [CA1046: Do not overload operator equals on reference types](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1046?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1050.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1050.md) | [CA1050: Declare types in namespaces](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1050?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1051.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1051.md) | [CA1051: Do not declare visible instance fields](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1051?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1052.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1052.md) | [CA1052: Static holder types should be Static or NotInheritable](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1052?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1054.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1054.md) | [CA1054: URI parameters should not be strings](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1054?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1055.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1055.md) | [CA1055: URI return values should not be strings](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1055?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1056.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1056.md) | [CA1056: URI properties should not be strings](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1056?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1062.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1062.md) | [CA1062: Validate arguments of public methods](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1062?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1710.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1710.md) | [CA1710: Identifiers should have correct suffix](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1710?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1802.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1802.md) | [CA1802: Use Literals Where Appropriate](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1802?branch=pr-en-us-35091) |
| [docs/fundamentals/code-analysis/quality-rules/ca1805.md](https://github.com/dotnet/docs/blob/90ff15a6c8f297a8c67d6eb8bfc15519af7af297/docs/fundamentals/code-analysis/quality-rules/ca1805.md) | [CA1805: Do not initialize unnecessarily.](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1805?branch=pr-en-us-35091) |

</details>

> **Note**
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/35091/checks).

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1000.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ dev_langs:
 | **Rule ID**                         | CA1000                       |
 | **Category**                        | [Design](design-warnings.md) |
 | **Fix is breaking or non-breaking** | Breaking                     |
+| **Enabled by default in .NET 7**    | No                           |
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca1000.md:25]: Are these changes automated?

**Review by BillWagner — APPROVED** [review]
I skimmed this, and it looks good to me.

Is there a tool that is automating this?

File: docs/fundamentals/code-analysis/quality-rules/ca1000.md (hunk 1 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca1000.md:25]: The table formatting is done using a VS Code extension (per file). The rest is all manual.

**Comment by gewarren** [comment]
> Is there a tool that is automating this?

The table formatting is done using a VS Code extension. The rest is all manual.

**Pull request merged**